spaceship/lib/spaceship/two_step_client.rb

```diff
@@ -40,7 +40,16 @@ def handle_two_step(response)
         device_id = result.match(/.*\t.*\t\((.*)\)/)[1]
         select_device(r, device_id)
       elsif r.body.kind_of?(Hash) && r.body["trustedPhoneNumbers"].kind_of?(Array) && r.body["trustedPhoneNumbers"].first.kind_of?(Hash)
-        handle_two_factor(r)
+        puts("Two Step Verification for account '#{self.user}' is enabled")
+        puts("Please select a phone number to verify your identity")
+
+        available = r.body["trustedPhoneNumbers"].collect do |current|
+          "#{current['numberWithDialCode']}\tSMS\t(#{current['id']})"
+        end
+        result = choose(*available)
+        device_id = result.match(/.*\t.*\t\((.*)\)/)[1]
+        response = select_number(r, device_id)
+        handle_two_factor(response)
```

Comment:
I noticed you let the normal `handle_two_factor()` handle the received code. While that might work right now, I noticed that this POSTs to `https://idmsa.apple.com/appleauth/auth/verify/trusteddevice/securitycode`. But the flow on the website POSTs to `https://idmsa.apple.com/appleauth/auth/verify/phone/securitycode` - not the `trusteddevice` being replaced with `phone`. 

It would probably be useful to add a second param `type` to the `handle_two_factor` method that can be `phone` or `trusteddevice` that is used in the URL to be POSTed to.